src/main/java/com/tabnine/TabNineCompletionContributor.java

```diff
@@ -172,7 +174,12 @@ TabNineProcess getProcOrPrintError() {
                 req.max_num_results = max_num_results;
                 req.region_includes_beginning = (begin == 0);
                 req.region_includes_end = (end == doc.getTextLength());
-                return proc.request(req);
+                
+                try {
+                    return AppExecutorUtil.getAppExecutorService().submit(() -> proc.request(req)).get(1, TimeUnit.SECONDS);
+                } catch (Exception e) {
+                    return null;
```

Comment:
Don't we want to handle exceptions here? 
If not then you might as well catch `Throwable` to cover yourself

Reply:
It is possible that we would like to add an event here, but I'm not adding events unless explicitly requested anymore (due to latest event bombarding mixpanel). @dimacodota what do you say?

Reply:
I think it would make more sense to add  sentry instead

Reply:
but for now, write it to log, and add spere case for TimeoutException

Reply:
Hmmm.... What do we do in case of timeout? Just return null? What will happen when we send the next request? I'm pretty sure stuff will break if we send a request before receiving a response.

Were you able to reproduce it? How?